cli.js

```diff
@@ -632,28 +624,19 @@ async function zkeyExportSolidityVerifier(params, options) {
 
     if (options.verbose) Logger.setLogLevel("DEBUG");
 
-    const templates = {};
+    const generator = options.plugin || "snarkjs-generate-solidity";
+    const { verifiers } = await import(generator);
 
-    if (await fileExists(path.join(__dirname, "templates"))) {
-        templates.groth16 = await fs.promises.readFile(path.join(__dirname, "templates", "verifier_groth16.sol.ejs"), "utf8");
-        templates.plonk = await fs.promises.readFile(path.join(__dirname, "templates", "verifier_plonk.sol.ejs"), "utf8");
-        templates.fflonk = await fs.promises.readFile(path.join(__dirname, "templates", "verifier_fflonk.sol.ejs"), "utf8");
-    } else {
-        templates.groth16 = await fs.promises.readFile(path.join(__dirname, "..", "templates", "verifier_groth16.sol.ejs"), "utf8");
-        templates.plonk = await fs.promises.readFile(path.join(__dirname, "..", "templates", "verifier_plonk.sol.ejs"), "utf8");
-        templates.fflonk = await fs.promises.readFile(path.join(__dirname, "..", "templates", "verifier_fflonk.sol.ejs"), "utf8");
-    }
-
-    const verifierCode = await zkey.exportSolidityVerifier(zkeyName, templates, logger);
+    const verifierCode = await zkey.exportVerifier(zkeyName, verifiers, logger);
 
     fs.writeFileSync(verifierName, verifierCode, "utf-8");
 
     return 0;
 }
 
 
-// solidity gencall <public.json> <proof.json>
-async function zkeyExportSolidityCalldata(params, options) {
+// zkey export calldata <public.json> <proof.json>
+async function zkeyExportCalldata(params, options) {
```

Comment:
This function needs to catch any error about unsupported protocols and show the user the plugin name and unsupported protocol name